plans/run.pp

```diff
@@ -0,0 +1,56 @@
+# Starts a Puppet agent run on the specified targets.
+# Note: This plan may cause issues when run in Puppet Enterprise.
+# @param targets The targets to start a Puppet agent run on.
+plan puppet_agent::run (
+  TargetSpec $targets
+) {
+  # Check which targets have the agent installed by checking
+  # the agent of the version. No point in trying to run the
+  # agent if it's not installed.
+  $version_results = run_task(
+    'puppet_agent::version',
+    $targets,
+    'Check for Puppet agent',
+    '_catch_errors' => true
+  )
+
+  # Filter targets by those that have an agent installed and
+  # those that don't. The puppet_agent::version task will return
+  # version:null for any targets that don't have an agent.
+  $agentless_results = $version_results.filter_set |$result| {
+    $result['version'] == undef
+  }
+
+  $agent_results = $version_results.filter_set |$result| {
+    $result['version'] != undef
+  }
+
+  # Create fail results for agentless targets.
+  $error_results = $agentless_results.map |$result| {
+    $err = {
+      '_error' => {
+        'msg'     => 'Puppet agent is not installed on the target.',
+        'kind'    => 'puppet_agent/agent-not-installed',
+        'details' => {}
+      }
+    }
+
+    Result.new($result.target, $err)
+  }
+
+  # Run the agent on all targets that have the agent installed.
+  $run_results = run_task(
+    'puppet_agent::run',
+    $agent_results.targets,
+    'Run Puppet agent',
+    '_catch_errors' => true
+  )
+
+  # Merge all of the results into a single ResultSet so each
+  # target has a result.
+  return ResultSet.new(
+    $version_results.error_set.results +
```

Comment:
This results in odd output when the version task fails:
```
lucy ~/githubs/bolt (GH-2051) $ be bolt plan run puppet_agent::run -t floaty
Starting: plan puppet_agent::run
Starting: Check for Puppet agent on simple-woodwind.delivery.puppetlabs.net
Finished: Check for Puppet agent with 1 failure in 0.99 sec
Finished: plan puppet_agent::run in 1.0 sec
Failed on simple-woodwind.delivery.puppetlabs.net:
  The task failed with exit code 1 and no output
Failed on simple-woodwind.delivery.puppetlabs.net:
  Puppet agent is not installed on the target.
Failed on 2 targets: simple-woodwind.delivery.puppetlabs.net,simple-woodwind.delivery.puppetlabs.net
Ran on 2 targets
```
I think we want to make `$agentless_result` just from results that succeeded the version task so that failures don't show up twice, and if you want to be really fancy modify the error message results to be more helpful.